numpy/_core/src/multiarray/abstractdtypes.h

```diff
@@ -81,6 +81,11 @@ npy_find_descr_for_scalar(
     PyObject *scalar, PyArray_Descr *original_descr,
     PyArray_DTypeMeta *in_DT, PyArray_DTypeMeta *op_DT);
 
+// Get the pointer to the PyArray_DTypeMeta for the type associated with the typenum.
+static inline PyArray_DTypeMeta *
+typenum_to_dtypemeta(enum NPY_TYPES typenum) {
+    return NPY_DTYPE(PyArray_DescrFromType(typenum));
+}
```

Comment:
Sorry to be obnoxious about this, but I think it makes more sense to put this in `dtypemeta.h` along with other helpers for dtypemetas. Also both `casts.cpp` and `asbtractdtypes.c` already include `dtypemeta.h`, so no need to mess with includes either.

Reply:
Not a problem at all, I appreciate the support in getting this right!

I also noticed that my first commit starts with `[MAINT]` and not `MAINT:`, so I'm going to fix that now too.